src/types.md

```diff
@@ -768,7 +111,8 @@ let x: Vec<_> = (0..10).collect();
   There should be a broader discussion of type inference somewhere.
 -->
 
-## Type parameters
+
+### Type parameters
```

Comment:
What do you think about moving it to `items/generics.md`.  It seems like it would be natural for it to live there.

Reply:
No, I would prefer them to live with all of the other types.

Reply:
I feel like it would be inconsistent to do that.  Currently everything that was moved to a separate page is an actual type.  To me, type parameters are not types themselves (they don't have a defined meaning on their own).  Similar to `Self` and `_`, they are things that can be resolved into an actual type.

Perhaps it could be moved as a second-level page under "Type System"?  The same question could be applied to `Inferred type` which probably equally deserves to be separate (considering it probably needs a lot more detail and examples).

Reply:
They don't resolve to an actual type until monomorphisation, so for almost all of compilation they are their own types with their own rules.

Either way, I'm happy to merge this as is once it has been rebased (again -_-).

Reply:
ok, I agree! 😄  I went ahead and moved the inferred type to its own page, too, since I foresee that potentially getting long by itself.  It's not too hard to move these around, and with the redirects it hopefully isn't too disruptive (although it's a little rough on outstanding PRs).